### ISSUE TITLE
PackageSignatureVerifier reports Valid without ever comparing the signed hash to the package bytes

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#3532]
## What

`PackageSignatureVerifier.VerifySignature` extracts the signed content hash and then returns `SignatureStatus.Valid` without ever comparing that hash against the actual `.nupkg` bytes.

The gap is acknowledged in the code (`src/NuGetFetch/PackageSignatureVerifier.cs`):

```csharp
// Parse the signed content hash for transparency.
// ...
// Full content hash verification (comparing this
// hash against the actual package bytes) requires implementing NuGet's
// SignedPackageArchiveUtility ZIP hashing algorithm, which manipulates raw
// ZIP bytes — not implemented in this lightweight client.
string? signedContentHash = TryExtractSignedHash(signedCms.ContentInfo.Content);
```

`signedContentHash` is only carried into the result as the `ContentHash` property. Nothing consumes it as a check.

## Why it matters

`CheckSignature(verifySignatureOnly: true)` proves the signature blob is internally consistent and was produced by the certificate holder. It says nothing about the package payload. So a `.nupkg` whose contents were altered while its intact original signature blob was retained passes every check here and is reported `SignatureStatus.Valid`.

That is the one thing a signature check exists to prevent, and `Valid` is a strong word for a result that does not establish it.

## Provenance

Pre-existing, inherited from upstream `nuget-fetch` and imported byte-identically by #3423, which deliberately makes no behavior changes ("an import is not an audit"). Filing separately so the import stays faithful and this is decided on its own merits.

## Options

1. Implement NuGet's `SignedPackageArchiveUtility` ZIP hashing and actually compare.
2. Keep the current depth but stop calling it `Valid` — report a status that distinguishes "signature is well-formed and trusted" from "package contents verified", so callers cannot read more into it than was checked.

Option 2 is cheap and removes the misleading claim; option 1 is the real fix.
